coral/src/app/features/topics/request/TopicEditRequest.test.tsx

```diff
@@ -0,0 +1,290 @@
+import { Context as AquariumContext } from "@aivenio/aquarium";
+import { cleanup, screen, waitFor, within } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { Route, Routes } from "react-router-dom";
+import TopicEditRequest from "src/app/features/topics/request/TopicEditRequest";
+import { getAllEnvironmentsForTopicAndAcl } from "src/domain/environment";
+import { mockedEnvironmentResponse } from "src/domain/environment/environment-api.msw";
+import { transformEnvironmentApiResponse } from "src/domain/environment/environment-transformer";
+import { editTopic, getTopicDetailsPerEnv } from "src/domain/topic";
+import { customRender } from "src/services/test-utils/render-with-wrappers";
+
+const mockedUsedNavigate = jest.fn();
+jest.mock("react-router-dom", () => ({
+  ...jest.requireActual("react-router-dom"),
+  useNavigate: () => mockedUsedNavigate,
+}));
+
+const mockedUseToast = jest.fn();
+jest.mock("@aivenio/aquarium", () => ({
+  ...jest.requireActual("@aivenio/aquarium"),
+  useToast: () => mockedUseToast,
+}));
+
+jest.mock("src/domain/topic/topic-api.ts");
+const mockEditTopic = editTopic as jest.MockedFunction<typeof editTopic>;
+const mockGetTopicDetailsPerEnv = getTopicDetailsPerEnv as jest.MockedFunction<
+  typeof getTopicDetailsPerEnv
+>;
+
+jest.mock("src/domain/environment/environment-api.ts");
+const mockGetAllEnvironmentsForTopicAndAcl =
+  getAllEnvironmentsForTopicAndAcl as jest.MockedFunction<
+    typeof getAllEnvironmentsForTopicAndAcl
+  >;
+
+const mockEnvironments = transformEnvironmentApiResponse(
+  mockedEnvironmentResponse
+).map((env) => ({
+  ...env,
+  params: {
+    ...env.params,
+    defaultPartitions: 2,
+    defaultRepFactor: 1,
+    maxPartitions: 2,
+    maxRepFactor: 1,
+  },
+}));
+
+const mockTopicDetails = {
+  topicExists: true,
+  topicId: "1200",
+  topicContents: {
+    topicName: "test-topic-name",
+    noOfPartitions: 2,
+    description: "Topic description",
+    noOfReplicas: "1",
+    teamId: 0,
+    showEditTopic: true,
+    showDeleteTopic: false,
+    topicDeletable: false,
+    advancedTopicConfiguration: {
+      "cleanup.policy": "compact",
+    },
+  },
+};
+
+describe("<TopicEditRequest />", () => {
+  const originalConsoleError = console.error;
+
+  beforeAll(() => {
```

Comment:
We should not use a `beforeAll` for all tests. This tests now all depend on each other and are not scoped clearly, so one assertion could fail because of something we did for another assertion, which can get very hard to debug. Adding new tests can also be harder later, because it may not be clear that the `it` blocks all affect each other.

Reply:
Yes! The first block ("Renders all fields with correct default values and disabled states") can be a `beforeAll` because that's the "we render it once and check different elements" without any user input, so we don't need to do the whole setup and teardown for every it and can improve performance (a liiiiiiiiiiiiiiittle bit).